src/components/QRDialog.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+	<div class="qrDialog__content">
+		<h2>{{ title }}</h2>
+		<div>
+			<img :src="uri"
+				:title="text"
+				:alt="t('forms', 'QR code representation of {text}', { text: text })">
+		</div>
+	</div>
+</template>
+
+<script>
+
+import QRCode from 'qrcode'
+
+export default {
+	name: 'QRDialog',
+
+	props: {
+		title: {
+			type: String,
+			default: '',
+		},
+		text: {
+			type: String,
+			default: '',
+		},
+	},
+
+	data() {
+		return {
+			uri: {
+				type: String,
+				default: '',
+			},
+		}
+	},
+
+	created() {
+		this.generateQr()
+	},
```

Comment:
```suggestion
	watch {
		text: {
			immediate: true,
			handler() {
				this.generateQr()
			},
		},
	},
```

Allow to be reactive to prop changes.